gm4_lively_lily_pads/data/gm4_lively_lily_pads/function/mechanics/interactions/candle/interact_rcd.mcfunction

```diff
@@ -0,0 +1,23 @@
+# candle right click
+# @s = player who right clicked on a candle
+# at @s
+# run from advancement: candle_rcd
+
+advancement revoke @s only gm4_lively_lily_pads:candle_rcd
+
+# flags and player data
+execute if entity @s[gamemode=adventure] run scoreboard players set $adventure gm4_llp.data 1
+execute store success score $holding_lighter gm4_llp.data if items entity @s weapon.mainhand #minecraft:creeper_igniters
```

Comment:
Don't use block tags not intended for your specific use-case, as Mojang may decide to add entries to these block tags that do not match your use case. Make your own instead.